android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIText.java

```diff
@@ -298,20 +302,15 @@ public void propertyChanged(String key, Object oldValue, Object newValue, KrollP
 		} else if (key.equals(TiC.PROPERTY_COLOR)) {
 			tv.setTextColor(TiConvert.toColor((String) newValue));
 		} else if (key.equals(TiC.PROPERTY_HINT_TEXT)) {
-			tv.setHint(TiConvert.toString(newValue));
+			int type = proxy.getProperties().getInt(TiC.PROPERTY_HINT_TYPE);
+			setHintText(type, TiConvert.toString(newValue));
 		} else if (key.equals(TiC.PROPERTY_HINT_TEXT_COLOR)) {
 			tv.setHintTextColor(TiConvert.toColor((String) newValue));
 		} else if (key.equals(TiC.PROPERTY_HINT_TYPE)) {
 			String hintText = TiConvert.toString(proxy.getProperty(TiC.PROPERTY_HINT_TEXT));
 			if (hintText != null) {
 				int type = TiConvert.toInt(newValue);
-				if (type == UIModule.HINT_TYPE_STATIC) {
-					textInputLayout.setHint("");
-					tv.setHint(hintText);
-				} else if (type == UIModule.HINT_TYPE_ANIMATED) {
-					tv.setHint("");
-					textInputLayout.setHint(hintText);
-				}
+				setHintText(type, hintText);
```

Comment:
Dynamically changing the "hintType" property ignores the "attributedHintText" property. Perhaps it should be written like this?

```java
Object text = d.get(TiC.PROPERTY_ATTRIBUTED_HINT_TEXT);
if (text instanceof AttributedStringProxy) {
	setAttributedStringHint((AttributedStringProxy)text);
} else {
	text = TiConvert.toString(d.get(TiC.PROPERTY_HINT_TEXT), "");
	if (text != null) {
		int type = TiConvert.toInt(newValue);
		setHintText(type, (String)text);
	}
}
```